File: Unique-Binary-Search-Trees/Unique-Binary-Search-Trees.py

class Solution:
    # @return an integer
    def numTrees(self, n):
        if n==0 or n==1:
            return 1
        elif n==2:
            return 2
        else:
            ans=0
            for i in range(n):
                ans+=self.numTrees(i)*self.numTrees(n-1-i)
            return ans
                
 

File: Unique-Binary-Search-Trees/Unique-Binary-Search-Trees.cpp

class Solution {
public:
    int numTrees(int n) {
        if(n==0)
            return 1;
        else if(n==1)
            return 1;
        else if(n==2)
            return 2;
        else
        {
            int ans=0;
            for(int i=0;i<n;i++)
                ans+=numTrees(i)*numTrees(n-i-1);
            return ans;
        }
    }
};


 